xinference/core/model.py

```diff
@@ -181,9 +199,27 @@ def __init__(
         }
         self._loop: Optional[asyncio.AbstractEventLoop] = None
 
+        self._scheduler_ref = None
+        self._isolation = None
+
     async def __post_create__(self):
         self._loop = asyncio.get_running_loop()
 
+        if self.allow_batching():
+            from ..isolation import Isolation
+            from .scheduler import SchedulerActor
+
+            self._scheduler_ref = await xo.create_actor(
+                SchedulerActor,
+                address=self.address,
+                uid=SchedulerActor.gen_uid(self.model_uid(), self._model.rep_id),
+            )
+            self._isolation = Isolation(asyncio.new_event_loop(), threaded=True)
+            self._isolation.start()
+            asyncio.run_coroutine_threadsafe(
+                self._scheduler_ref.run(), loop=self._isolation.loop
```

Comment:
Why not call the `run()` in the `__post_create__` of `SchedulerActor`? I think that could be cleaner.